src/OrchardCore.Modules/OrchardCore.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -391,19 +391,23 @@ private async Task GenerateContainerPathFromPattern(AutoroutePart part)
                     ContentItem = part.ContentItem
                 };
 
-                // We keep the current culture as a reference for later
-                var currentCulture = CultureInfo.CurrentUICulture;
-
                 _contentManager ??= _serviceProvider.GetRequiredService<IContentManager>();
                 var cultureAspect = await _contentManager.PopulateAspectAsync(part.ContentItem, new CultureAspect());
 
-                CultureInfo.CurrentUICulture = cultureAspect.Culture;
-
-                part.Path = await _liquidTemplateManager.RenderAsync(pattern, NullEncoder.Default, model,
-                    scope => scope.SetValue("ContentItem", model.ContentItem));
+                // We keep the current culture as a reference for later
+                var currentCulture = CultureInfo.CurrentUICulture;
 
-                // We reassign the proper culture to the current execution flow
-                CultureInfo.CurrentUICulture = currentCulture;
+                CultureInfo.CurrentUICulture = cultureAspect.Culture;
```

Comment:
I wanted to make sure the whole process won't throw. Though here the finally is only necessary if you actually change the CultureINfo.CurrentUICulture so that makes sense.

Reply:
We should probably also make sure that cultureAspect.Culture is never null.

Reply:
Yes you're right if it could be null

@hishamco will have to take it into account in its PR if it updates this part

Reply:
The PR that changes the `await orchardHelper.GetContentCultureAsync` might do that for when we want to index with Lucene. Though we don't use `await orchardHelper.GetContentCultureAsync` here either and we should probably too.

Reply:
I think we are due to fix those issues all at once else we do small bits here and there and it is not consistent.

Reply:
Right now cultureAspect.Culture can't be null because of the default of the CultureAspect.